orderbook/src/orderbook.rs

```diff
@@ -51,6 +53,25 @@ impl OrderBook {
             Err(RemoveOrderError::DoesNotExist)
         }
     }
+
+    // Clean up operations like removing expired orders.
+    pub async fn run_maintenance_forever(&self) -> ! {
+        loop {
+            tracing::debug!("running order book maintenance");
+            self.remove_expired_orders(SystemTime::now()).await;
```

Comment:
I think it would be good to separate out the way we decide when to do maintenance (loop with sleep vs. reacting on newBlockEvents stream) from the actual maintenance we are doing (removing expired orders, checking for updated allowances, etc).

Could there be a maintenance scheduler that just invokes orderbook.do_maintenance?

Reply:
That's definitely something we could do but I don't want to over engineer now. At this point it just a sleep now so introducing a maintenance scheduler is overkill.

Reply:
Can we move the `run_maintenance_forever` outside of orderbook into a separate place (I don't think it uses self atm). The orderbook will do a lot of things related to the orderbook data itself so scheduling the maintenance should really live somewhere else (I'm fine with leaving it a function for now).

Reply:
Makes sense. I moved it into main since that is the only place where we use tokio directly to create tasks. If that function gets more complicated it can also go into it's own module.